python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -246,6 +246,14 @@ def is_empty(self) -> bool:
     def empty() -> "TimeWindow":
         return TimeWindow(start=datetime.max, end=datetime.max)
 
+    @lru_cache(maxsize=1000)
+    def start_timestamp(self) -> float:
+        return self.start.timestamp()
```

Comment:
i wonder how often we end up redundantly caching start & end values across `TImeWIndow`s with the same `datetime` values but not the exact same start & end combo

Reply:
interesting...

Reply:
I used a shared cache for both - not seeing a ton of difference in practice but I certainly agree with your intuition here